tests/unit/Domain/Service/ContractualBaseServiceTest.php

```diff
@@ -55,6 +55,27 @@ public function testWriteContractualBase(
         $this->assertEquals($expectedContractualBase, $entity->getMetaData()->getCoin()->getContractualBase());
     }
 
+    public function testContractualBaseForTest()
+    {
+        $entity = $this->createMockEntity(
+            Constants::ENVIRONMENT_TEST,
+            Constants::TYPE_SAML,
+            Constants::SERVICE_TYPE_INSTITUTE
+        );
+        $this->assertNull($entity->getMetaData()->getCoin()->getContractualBase());
+
+        $pristine = $this->createMockEntity(
+            Constants::ENVIRONMENT_TEST,
+            Constants::TYPE_SAML,
+            Constants::SERVICE_TYPE_INSTITUTE,
+        );
+        $pristine->getMetaData()->getCoin()->setContractualBase(Constants::CONTRACTUAL_BASE_IX);
+
+        $this->service->writeContractualBaseForTestEntity($entity, $pristine);
+
+        $this->assertSame(Constants::CONTRACTUAL_BASE_IX, $entity->getMetaData()->getCoin()->getContractualBase());
+    }
```

Comment:
Nice test, personally I would add some coverage for the non-happy flows. 

Testing that contractual base is not set when:
- ENV=prod
- The pristine entity does not have the contractual base coin

Reply:
added